src/bundlr.rs

```diff
@@ -1,17 +1,20 @@
-use serde::Deserialize;
-use serde_json::Value;
+use std::collections::HashMap;
 
+use crate::currency::Currency;
 use crate::error::BundlrError;
 use crate::tags::Tag;
 use crate::{signers::signer::Signer, BundlrTx};
+use num_bigint::BigUint;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
 
 #[allow(unused)]
-pub struct Bundlr<T> {
-    url: String,
-    chain: String,
-    currency: String,
-    signer: T,
+pub struct Bundlr<'a> {
+    url: String, // FIXME: type of this field should be Url
+    currency: Currency,
+    signer: &'a dyn Signer,
```

Comment:
Why not go a bit further and drop the dynamic dispatch here:
```rust
pub struct Bundlr<'a, S> where S: Signer {
    url: String, // FIXME: type of this field should be Url
    currency: Currency,
    signer: &'a S,
    client: reqwest::Client,
    pub_info: PubInfo,
}
```

Reply:
Or maybe drop that `where S: Signer` and add it as a requirement where it's really needed.

Reply:
I think reducing the arguments as minimal as possible is the best approach, so we can do::

```rust
let bundlr = Bundlr::new( ... );
```

Instead of

```rust
let bundlr = Bundlr::<ArweaveSigner>::new( ... );
let bundlr = Bundlr::<dyn Signer>::new( ... );
```